### PR TITLE
Updates for upload tool

### PR DESCRIPTION
**Pull request by brentru**
- Adds `esptool` object for firmware upload tool, includes offsets and flash size per-board.

- Renames ESP32/8266 board folders to match names within the ci-arduino generation tool (https://github.com/adafruit/ci-arduino/blob/master/build_platform.py#L62), and firmware names.

- Removes `index.json` file

### CODE REVIEW

File: boards/feather_esp32/definition.json
Comment (lorennorman) [inline boards/feather_esp32/definition.json:12]: These should not be strings, just decimal numbers. Unless we have a reason to use strings here that i'm not aware of.

File: boards/feather_esp32/definition.json
Comment (lorennorman) [inline boards/feather_esp32/definition.json:15]: this is an object nested inside an array, should just be an object, i believe (remove the wrapping `[]`)

**Review by lorennorman — CHANGES_REQUESTED** [review]
both of my comments repeat to the next 2 board defs

File: boards/feather_esp32/definition.json
Comment (brentru) [reply boards/feather_esp32/definition.json]: Resolved in https://github.com/adafruit/Wippersnapper_Boards/pull/39/commits/7eddb8f8999a502bf8887d96d3febcef8f2cfec2

File: boards/feather_esp32/definition.json
Comment (lorennorman) [inline boards/feather_esp32/definition.json:19]: All 3 boards in this PR need the "structure" property filled out with the real filenames from the firmware releases.

Note, make sure all referenced files are in the release `.zip`, I notice that `boot_app0.bin` isn't in the esp32 release zip, for example.

Comment (brentru) [reply boards/feather_esp32/definition.json]: Resolved in https://github.com/adafruit/Wippersnapper_Boards/pull/39/commits/31fb217f05bfffe90bd29590216045d23c7a89b2 and https://github.com/adafruit/Adafruit_Wippersnapper_Arduino/pull/227

**Review by ladyada — APPROVED** [review]

**Comment by brentru** [comment]
**NOTE: Holding off on this until approved by Loren**

**Pull request merged**
